tests/server/server.spec.ts

```diff
@@ -1,21 +1,38 @@
 import Server from '../../src/server/server'
-
 let hasMockClientConnected = false
 
 jest.mock('net', () => {
   type MaybeNumber = number | undefined
   type MaybeString = string | undefined
   type MaybeVoidFunction = (() => void) | undefined
-  type SocketCallback = ((socket: any) => void) | undefined
+  type MaybeServerCallback = ((socket: any) => void) | undefined
+  type StringOrUint8Arr = string | Uint8Array
+  type WriteFunction = (data: StringOrUint8Arr) => boolean
+  type SocketEventEmitter = (
+    eventName: string,
+    eventListener: (data: Buffer) => void
+  ) => void
+  type EndFunction = (
+    buffer: StringOrUint8Arr,
+    callback: MaybeVoidFunction
+  ) => void
```

Comment:
🎉 Nice, good job keeping up with the types!

Reply:
Thanks for introducing me to the type aliases, Naomi! Really does help me with simplifying things.